HW4/HW4.py

```diff
@@ -0,0 +1,381 @@
+import sys
+
+import numpy
+import pandas as pd
+import statsmodels.api
+from plotly import express as px
+from plotly import figure_factory as ff
+from plotly import graph_objects as go
+from plotly.subplots import make_subplots
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import confusion_matrix
+
+
+def main():
+    """Set the arguments below until I can get argparse in here"""
+    # arguments: df, response (string), predictors (list)
+    # testing data only
+    # names = ["Sepal Length", "Sepal Width", "Petal Length", "Petal Width", "Class"]
+    # df = pd.read_csv("iris.data", names=names)
+    # df['boo_response']=df['Class'].apply(lambda x: True if x=='Iris-setosa' else False)
+
+    # Hey user, set these until I get argparse in here
+    df = pd.DataFrame()
+    response = ""
+    predictors = []
+    # predictor = 'Sepal Length'
+
+    # testing only
+    # create boolean response - probably should remove
+    # df['boo_response']=df['Class'].apply(lambda x: True if x=='Iris-setosa' else False)
+
+    # steps
+    # Determine if response is continuous or boolean
+    y = response
+    if df[response].unique().size > 2:
+        response_type = "continuous"
+    else:
+        # hard coding to boolean rather than categorical
+        response_type = "boolean"
+
+    # Determine if predictors are categorical or continuous
+    cont_master_list = df._get_numeric_data().columns.to_list()
+    predictor_types = {}
+    cont_predictor_col_names = []
+    for predictor in predictors:
+        predictor_type = ""
+        if predictor in cont_master_list:
+            predictor_type = "continuous"
+        if True in df[predictor].unique().tolist():
+            predictor_type = "boolean"
+        if predictor_type == "":
+            if df.dtypes[predictor] == "O":
+                predictor_type = "categorical"
+        if predictor_type == "":
+            predictor_type = input(
+                f"Toss me a bone, what's the predictor type for {predictor}? "
+                f"(continuous, categorical, or boolean):"
+            )
+        if predictor_type == "":
+            print("that's not very nice")
+            predictor_type = "categorical"
+        predictor_types[predictor] = predictor_type
+        if predictor_type == "continuous":
+            cont_predictor_col_names.append(predictor)
+
+    # Heatplot
+    def Heatplot(x):
+        conf_matrix = confusion_matrix(df[x].astype(str), df[y].astype(str))
+        fig_cat_cat = go.Figure(
+            data=go.Heatmap(z=conf_matrix, zmin=0, zmax=conf_matrix.max())
+        )
+        fig_cat_cat.update_layout(
+            title="",
+            xaxis_title="Response",
+            yaxis_title="Predictor",
+        )
+        filename = x + " heatmap.html"
+        fig_cat_cat.write_html(file=filename, include_plotlyjs="cdn")
+        return filename
+
```

Comment:
I like the way you save the filename here, I should do this as well if we can figure out the hyperlink for local files into a pandas df